[nx-plugin] TS5110: Option 'module' must be set to 'NodeNext' when option 'moduleResolution' is set to 'NodeNext'.

### Current Behavior

E2E target of plugins created in Nx 20.2.0-beta.4, fails with the following error:

```
nx run my-plugin-e2e:e2e

 NX   Jest: Failed to parse the TypeScript config file ~/Projects/sandbox/packages/my-plugin-e2e/jest.config.ts

  TSError: ⨯ Unable to compile TypeScript:
error TS5110: Option 'module' must be set to 'NodeNext' when option 'moduleResolution' is set to 'NodeNext'.

Pass --verbose to see the stacktrace.
```

### Expected Behavior

e2e target should run without issues

### GitHub Repo

_No response_

### Steps to Reproduce

1. Create a empty workspace with Nx 20.2.0-beta.4 and use the TS solution
2. Create a new plugin with e2e tests
3. Run the e2e target


### Nx Report

```shell
npx nx report         

 NX   Report complete - copy this into the issue template

Node           : 22.11.0
OS             : darwin-x64
Native Target  : x86_64-macos
npm            : 10.9.1

nx                 : 20.2.0-beta.4
@nx/js             : 20.2.0-beta.4
@nx/jest           : 20.2.0-beta.4
@nx/eslint         : 20.2.0-beta.4
@nx/workspace      : 20.2.0-beta.4
@nx/devkit         : 20.2.0-beta.4
@nx/eslint-plugin  : 20.2.0-beta.4
@nx/plugin         : 20.2.0-beta.4
@nx/vite           : 20.2.0-beta.4
@nx/web            : 20.2.0-beta.4
typescript         : 5.5.4
---------------------------------------
Registered Plugins:
@nx/js/typescript
@nx/eslint/plugin
@nx/jest/plugin
```

### Failure Logs

```shell

```

### Package Manager Version

npm 10.9.1

### Operating System

- [x] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

nx run my-plugin-e2e:e2e

 NX   Jest: Failed to parse the TypeScript config file ~/Projects/sandbox/packages/my-plugin-e2e/jest.config.ts

  TSError: ⨯ Unable to compile TypeScript:
error TS5110: Option 'module' must be set to 'NodeNext' when option 'moduleResolution' is set to 'NodeNext'.

Pass --verbose to see the stacktrace.

--- a/packages/jest/src/executors/jest/jest.impl.ts
+++ b/packages/jest/src/executors/jest/jest.impl.ts
@@ -21,6 +21,10 @@ export async function jestExecutor(
   options: JestExecutorOptions,
   context: ExecutorContext
 ): Promise<{ success: boolean }> {
+  // Jest registers ts-node with module CJS https://github.com/SimenB/jest/blob/v29.6.4/packages/jest-config/src/readConfigFileAndSetRootDir.ts#L117-L119
+  // We want to support of ESM via 'module':'nodenext', we need to override the resolution until Jest supports it.
+  process.env.TS_NODE_COMPILER_OPTIONS ??= '{"moduleResolution":"node10"}';
+
   const config = await jestConfigParser(options, context);
 
   const { results } = await runCLI(config, [options.jestConfig]);

